autoware_adapi_v1_msgs/package.xml

```diff
@@ -2,23 +2,29 @@
 <?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
 <package format="3">
   <name>autoware_adapi_v1_msgs</name>
-  <version>0.0.0</version>
-  <description>The autoware_adapi_v1_msgs package</description>
+  <version>1.1.0</version>
+  <description>The Autoware ADAPI interfaces</description>
   <maintainer email="isamu.takagi@tier4.jp">Takagi, Isamu</maintainer>
+  <maintainer email="yutaka.kondo@tier4.jp">Yutaka Kondo</maintainer>
   <license>Apache License 2.0</license>
 
   <buildtool_depend>ament_cmake_auto</buildtool_depend>
 
+  <build_depend>builtin_interfaces</build_depend>
```

Comment:
What is the reason for replacing `depend` tag? We can use it as long as `build` and `exec` do not have separate dependencies. And for this package, both are required.

Reply:
As a best practice for writing dependency in `package.xml`, it is better to define individual `build_depend`, `build_export_depend`, `exec_depend`, and `test_depend` tags to minimize the need to use the `depen tag directly.
This will allow for the installation of the minimum necessary number of dependencies.
In this case, I think we don't need `build_export_depend` and `test_depend` so far.

Reply:
https://docs.ros.org/en/humble/Tutorials/Intermediate/Rosdep.html#a-little-about-package-xml-files
According to this page:
- Basically, use `depend`.
- Separate if `build_depend` and `exec_depend` are different.
- Add `build_export_depend` if `build_depend` is used and there are indirect dependencies.
- Add `test_depend` when the test requires additional dependencies.

> I think we don't need build_export_depend and test_depend so far.

For message packages, `build_export_depend` is also needed because the headers of dependent packages are included indirectly. Also, semantically `depend` does not include `test_depend`.

ROS 2 common_interface also uses `depend`.
https://github.com/ros2/common_interfaces/blob/rolling/nav_msgs/package.xml

Reply:
Oh, my knowledge was old. Thank you for your feedback.
Reverted. https://github.com/autowarefoundation/autoware_adapi_msgs/pull/55/commits/f4c77c4f71d520d586922f26ab28c9e1b412de7e